### PR TITLE
CLI based searcher

### PR DESCRIPTION
**Pull request by buffalu**
- Adds a CLI-based searcher. Easy to explore the API, subscriptions, and send bundle.
- Rework auth to be inline vs background thread.

TODO:
- [ ] add instructions + readme

### CODE REVIEW

File: cli/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,279 @@
+use clap::{Parser, Subcommand};
+use env_logger::TimestampPrecision;
+use futures_util::StreamExt;
+use jito_protos::bundle::Bundle;
```
Comment (segfaultdoc) [inline cli/src/main.rs:4]: fmt?

File: searcher_service_client/src/client_with_auth.rs [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+                self.refresh_token = Some(refresh_token.value);
+                let refresh_token_expiration = refresh_token.expires_at_utc.unwrap();
+                self.refresh_token_expiration_time = Duration::new(
+                    refresh_token_expiration.seconds as u64,
+                    refresh_token_expiration.nanos as u32,
+                );
+            }
+            (_, true) => {
+                info!("refreshing access token");
+                let access_token =
```
Comment (buffalu) [inline searcher_service_client/src/client_with_auth.rs:62]: note: if this fails we might wanna try the entire re-auth workflow

File: backrun/src/main.rs [hunk 3]
```diff
@@ -50,11 +50,7 @@ use tonic::{Response, Status};
 struct Args {
     /// Address for auth service
     #[clap(long, env)]
-    auth_addr: String,
-
-    /// Address for searcher service
-    #[clap(long, env)]
-    searcher_addr: String,
+    block_engine_addr: String,
```
Comment (buffalu) [inline backrun/src/main.rs:53]: todo: make sure readme is updated

File: cli/src/main.rs [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+    /// Prints out information on connected leaders
+    ConnectedLeaders,
+    /// Prints out information on the tip accounts
+    GetTipAccounts,
+    /// Sends a 1 lamport bundle
+    SendBundle {
+        /// RPC URL
+        #[clap(long, required = true)]
+        rpc_url: String,
+
```
Comment (buffalu) [inline cli/src/main.rs:65]: remove line

**Review by segfaultdoc — APPROVED** [review]

File: cli/src/main.rs (hunk 1 above)
Comment (buffalu) [reply cli/src/main.rs:4]: no rustfmt here. will add

File: backrun/Cargo.toml [hunk 5]
```diff
@@ -17,10 +17,10 @@ log = "0.4.14"
 prost-types = "0.8.0"
 rand = "0.8.5"
 searcher-service-client = { path = "../searcher_service_client" }
-solana-client = "=1.11.2"
-solana-metrics = "=1.11.2"
-solana-sdk = "=1.11.2"
-solana-transaction-status = "=1.11.2"
+solana-client = "=1.14.13"
+solana-metrics = "=1.14.13"
+solana-sdk = "=1.14.13"
```
Comment (jedleggett) [inline backrun/Cargo.toml:22]: do we need jito-solana here to get bundle result errors?

Comment (jedleggett) [reply backrun/Cargo.toml:22]: nvm, the message is parsed in the block engine (searcher_service) and sent here as string

**Pull request merged**